x-pack/plugins/serverless_security/public/components/get_started/toggle_panel.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useReducer } from 'react';
+import { EuiEmptyPrompt, EuiFlexGroup, EuiFlexItem, useEuiShadow, useEuiTheme } from '@elastic/eui';
+
+import { css } from '@emotion/react';
+
+import {
+  Switch,
+  TogglePanelAction,
+  ProductId,
+  TogglePanelReducer,
+  ToggleStepAction,
+  GetStartedPageActions,
+  StepId,
+  CardId,
+} from './types';
+import * as i18n from './translations';
+import { ProductSwitch } from './product_switch';
+import { useSetUpCardSections } from './use_setup_cards';
+import { useStorage } from './use_storage';
+import { useKibana } from '../../services';
+
+const reducer = (state: TogglePanelReducer, action: TogglePanelAction | ToggleStepAction) => {
+  if (action.type === GetStartedPageActions.ToggleSection) {
+    if (state.activeSections.has(action.payload?.section)) {
+      state.activeSections.delete(action.payload?.section);
+    } else {
+      state.activeSections.add(action.payload?.section);
+    }
+
+    return {
+      ...state,
+      activeSections: new Set([...state.activeSections]),
+    };
+  }
```

Comment:
We should not mutate the current state in a reducer. We can create a `new Set` first:
```suggestion
  if (action.type === GetStartedPageActions.ToggleSection) {
    const activeSections = new Set([...state.activeSections]);
    if (activeSections.has(action.payload.section)) {
      activeSections.delete(action.payload.section);
    } else {
      activeSections.add(action.payload.section);
    }
    return { ...state, activeSections };
  }
```
Same thing for the _AddFinishedStep_ case.